### ISSUE TITLE
Missing query escape in jobs

### ISSUE THREAD

**Issue by peternijssen** [xanzy/go-gitlab#253]
Hi,

I noticed that the Tags.GetTag() does a URL call like this;
```
/api/v3/projects/xxx%2Fyyy/repository/branches/dev
```
while Jobs. GetJobArtifacts() does a URL call like this;
```
/api/v3/projects/xxx/yyy/jobs/artifacts/dev/download?job=build
```

I suppose the difference is because there is no consistent query escape being done;
https://github.com/xanzy/go-gitlab/blob/55af103d769f3715bf6dddfdea30ce2899fb0e8b/tags.go#L85
https://github.com/xanzy/go-gitlab/blob/55af103d769f3715bf6dddfdea30ce2899fb0e8b/jobs.go#L157

I suggest to make this consistent. In my case, only the escaped ones (Tags) work properly.

**Comment by svanharmelen** [comment]
Oeps... That is indeed a bug. `project` should always be escaped...

**Comment by svanharmelen** [comment]
Fixed by commit b1362a3ba2b24ba59f92c2027f4d02142346dfdd
